Sources/MessagingPush/MessagingPushConfigOptions.swift

```diff
@@ -0,0 +1,21 @@
+public struct MessagingPushConfigOptions {
+    // Used to create new instance of MessagingPushConfigOptions when the MessagingPushAPN module is configured.
+    // Each property of the MessagingPushConfigOptions object can be modified by the user.
+    public enum Factory {
+        public static func create() -> MessagingPushConfigOptions {
+            MessagingPushConfigOptions(
+                autoFetchDeviceToken: true
+            )
+        }
+    }
+
+    public enum Keys: String { // Constants used to map each of the options in MessagingPushConfigOptions
+        case autoFetchDeviceToken
+    }
+
+    /**
+     Enable automatic fetching of device token by the SDK without the need to write custom code by the customer.
+     On fetching the token, SDK will auto-register the device. This value is `true` by default.
+     */
+    public var autoFetchDeviceToken: Bool
+}
```

Comment:
```suggestion
public struct MessagingPushConfigOptions {
    /**
     Enable automatic fetching of device token by the SDK without the need to write custom code by the customer.
     On fetching the token, SDK will auto-register the device. This value is `true` by default.
     */
    public var autoFetchDeviceToken: Bool = true 
}
```

I think this struct can be simplified to provide the default value by assigning it to the property directly.

Reply:
This change will require code change in FCM and APN implementation as both modules use the same logic. There could be possibly other keys as well to be added and I believe we should let this as is as this is a tested logic in use for other configurations as well.